### PR TITLE
Feat/reader

### PR DESCRIPTION
**Pull request by dospore**
Add reader with two helper functions. This is basically just to make calculating the current apr easier. You can either use getCurrentCycleInfo with an expected rewards amount or getCycleInfo to get a previous cycles rewards amount. This info can be multiplied by the current eth price and mycPrice to calculate apr

- getCurrentCycleInfo gets the rewardsPerShare, supply and assets for the current cycle. The rewardsPerShare is based on a supplied ethRewardAmount. This can be used in conjunction with getCycleInfo which will use the previous cycles eth rewards to estimate a current cycle reward. Or you can pass it any arbitrary reward amount
- getCycleInfo will retrieve the info related to a given cycle. If you pass a cycle higher than the current cycle it will just get current cycle info

### CODE REVIEW

File: src/peripherals/LentMycReader.sol [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    /// @notice External function to access internal _getCurrentCycleInfo
+    function getCurrentCycleInfo(address lentMyc_, uint256 ethRewards) external view returns (uint256[] memory) {
+        return _getCurrentCycleInfo(lentMyc_, ethRewards);
+    }
+
+    /// @notice Get the current cycles reward reward and info given an eth reward amount
+    /// @dev Dospore
+    /// @param lentMyc_ address of lentMyc
+    /// @param ethRewards amount of eth that will be rewarded to current cycle stakers
+    /// @return Current cycles reward info based on the given eth rewards amount
```
Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:23]: Switch natspecs to use /** */ comments pls :pleading_face:

Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:23]: Change natspecs to use `/** */` comments please :pleading_face: 

Also full sentences please. Sorry for being annoying, just keeping styling consistent with the main contracts.

File: src/peripherals/LentMycReader.sol [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        info[0] = currentCyclesEthRewardsPerShare;
+        info[1] = currentCycleSupply;
+        info[2] = currentCycleMyc;
+
+        return info;
+    }
+
+
+    /// @notice Get an arbitrary cycles info
+    /// @dev Dospore
```
Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:45]: the `@dev` natspec tag is to explain to a developer any extra details.

File: src/peripherals/LentMycReader.sol [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        uint256[] memory info;
+        info[0] = currentCyclesEthRewardsPerShare;
+        info[1] = currentCycleSupply;
+        info[2] = currentCycleMyc;
+
+        return info;
+    }
+
+
+    /// @notice Get an arbitrary cycles info
```
Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:44]:
```suggestion
    /// @notice Get an arbitrary cycle's info
```

File: src/peripherals/LentMycReader.sol [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        if (fetchingCurrentCycle) {
+            uint256 lastCyclesEthRewards = cyclesEthRewardsPerShare.mulWadDown(cycleSupply);
+            return _getCurrentCycleInfo(lentMyc_, lastCyclesEthRewards);
+        }
+
+        uint256[] memory info;
+        info[0] = cyclesEthRewardsPerShare;
+        info[1] = cycleSupply;
+        info[2] = cycleMyc;
+
```
Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:74]: Would you be able to make it a struct to make it easier to guarantee correct ordering?

File: src/peripherals/LentMycReader.sol [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            cycle = cycle_ - 1;
+            fetchingCurrentCycle = true;
+        }
+
+        uint256 previousCumaltiveEthRewardsPerShare = lentMyc.cycleCumulativeEthRewards(cycle - 1);
+        uint256 cyclesCumaltiveEthRewardsPerShare = lentMyc.cycleCumulativeEthRewards(cycle);
+        uint256 cyclesEthRewardsPerShare = previousCumaltiveEthRewardsPerShare - cyclesCumaltiveEthRewardsPerShare;
+
+        (uint256 cycleSupply, uint256 cycleMyc) =  lentMyc.cycleSharesAndAssets(cycle);
+
```
Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:63]:
```suggestion
```

File: src/peripherals/LentMycReader.sol [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    /// @dev Dospore
+    /// @param lentMyc_ address of lentMyc
+    /// @param cycle number to retrieve info on
+    /// @return Given cycles reward info or the currentCycles info based on the previous cycle
+    function getCycleInfo(address lentMyc_, uint256 cycle) external view returns (uint256[] memory) {
+        ILentMyc lentMyc = ILentMyc(lentMyc_);
+        uint256 cycle_ = lentMyc.cycle();
+        bool fetchingCurrentCycle = false;
+        if (cycle >= cycle_) {
+            cycle = cycle_ - 1;
```
Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:54]:
```suggestion
        uint256 currentCycle = lentMyc.cycle();
        bool fetchingCurrentCycle = false;
        if (cycle >= currentCycle) {
            cycle = currentCycle - 1;
```

File: src/peripherals/LentMycReader.sol [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+contract LentMycReader {
+    using FixedPointMathLib for uint256;
+
+    /// @notice External function to access internal _getCurrentCycleInfo
+    function getCurrentCycleInfo(address lentMyc_, uint256 ethRewards) external view returns (uint256[] memory) {
+        return _getCurrentCycleInfo(lentMyc_, ethRewards);
+    }
+
+    /// @notice Get the current cycles reward reward and info given an eth reward amount
+    /// @dev Dospore
```
Comment (CalabashSquash) [inline src/peripherals/LentMycReader.sol:20]: the `@dev` natspec tag is to explain to a developer any extra details.

**Review by CalabashSquash — CHANGES_REQUESTED** [review]

File: src/peripherals/LentMycReader.sol (hunk 1 above)
Comment (dospore) [reply src/peripherals/LentMycReader.sol]: I added the natspecs and will update the sentences. But what do you mean by full sentences just like more explanation?

**Review by CalabashSquash — APPROVED** [review]

**Pull request merged**